### PR TITLE
resource_type: adapt to vocabulary transition

### PR DESCRIPTION
**Pull request by fenekku**
- closes https://github.com/inveniosoftware/invenio-app-rdm/issues/840 in part
- releases react-invenio-deposit line for May

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -1,5 +1,11 @@
 # Changes
 
+Version 0.14.0 (released 2021-05-19)
+
+* resource_type is now a vocabulary field (i.e. of the shape {id: <...>}),
+  so changes are made to accomodate this
+* mark May release
```
Comment (ppanero) [inline CHANGES.md:7]: for consistency with previous releases

```suggestion
* Mark May release
```

**Review by ppanero — COMMENTED** [review]
LGTM, but maybe someone more react-proficient can take a look @zzacharo @ntarocco

File: src/lib/components/ResourceTypeField.js [hunk 2]
```diff
@@ -23,111 +20,73 @@ export class ResourceTypeField extends Component {
...
+        }
+      }
+    );
+    return options;
   }
 
-  renderResourceTypeField = ({ field, form }) => {
-    const {
+  render() {
+    let {
```
Comment (ntarocco) [inline src/lib/components/ResourceTypeField.js:72]: Please use `const` whenever possible and not `let`.
```suggestion
    const {
```

**Review by ntarocco — APPROVED** [review]

File: src/lib/components/ResourceTypeField.js [hunk 3]
```diff
@@ -23,111 +20,73 @@ export class ResourceTypeField extends Component {
...
+   * @param {array} propsOptions - back-end options
+   * @returns {array} front-end options
+   */
   createOptions = (propsOptions) => {
-    let options = [];
-    for (const type of propsOptions.type) {
-      const subtypes = propsOptions.subtype.filter(
-        (e) => e['parent-value'] === type.value
-      );
+    let options = propsOptions.slice();
```
[the remark this answers is not in the document]
Comment (fenekku) [reply src/lib/components/ResourceTypeField.js:42]: (and removed the comment)

**Comment by fenekku** [comment]
Comments addressed thanks @ppanero  and @ntarocco . Merging when the tests pass.

**Pull request merged**